### ISSUE TITLE
CLI Tools Need Dynamic Params (query, body,env)

### ISSUE THREAD

**Issue by guotingchao** [rangav/thunder-client-support#1184]
** Please describe feature/problem details and solution you'd like.**

**When I use `$tc requestId-env 'someParamsName' to pass parameters`,
 I want to see if I can pass parameters dynamically so that I can implement batch interface test processing locally through the command line.**

** Describe alternatives you've considered**

For the time being, no method has been found through Document.

** Your Team Size Using TC:**
<= 5

**Comment by rangav** [comment]
Hi @guotingchao 

when you select Run Collection in the extension, you can select input file for the env variables used to pass data dynamically 

or you can use custom scripting to set data in pre request 
https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#custom-filters

**Comment by guotingchao** [comment]
> Hi @guotingchao
> 
> when you select Run Collection in the extension, you can select input file for the env variables used to pass data dynamically
> 
> or you can use custom scripting to set data in pre request https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#custom-filters

Thank you very much for your reply !
If possible, you'd like to use the parameter `tc 'requestId' --env 'Staging' -setEnv a='123'` in @ thunderclient/cli to set some variables to the env of the corresponding scope, which may make it more efficient for users to set values for environment variables in a command-line environment. 😄

**Comment by guotingchao** [comment]
🤔️ Just like using env in `Postman/Cli` to set the parameter pair of environment variables, this is a familiar mode of operation. 
```bash 
  --env-var "[environment-variable-name]=[environment-variable-value]"
```

**Comment by rangav** [comment]
Thanks for the feedback, Will review it.

Alternatively:
You can also open an environment in extension and quickly change values and run command in cli

**Comment by rangav** [comment]
Hi @guotingchao this feature will be part of next update.

the format will be as below
- `tc "reqIdOrName" --var-data ="name1=value1,name2=value2"`
- `tc --col "Customer" --var-data ="name1=value1,name2=value2"`

Please let me know your feedback.

**Comment by rangav** [comment]
This feature is now implemented and published to the marketplace, please update to `v2.9.0`

See all features released
https://github.com/rangav/thunder-client-support/releases/tag/v2.9.0

Please let me know your feedback.
